### PR TITLE
Fix: #901

### PR DESCRIPTION
**Pull request by SkullzOTS**

### CODE REVIEW

**Comment by kokekanon** [comment]
It works, at least in versions that use .dat and .spr.

In protobuf it doesn't work, although it’s not directly related to 901, but it’s something to keep in mind for the future.

8.6
image
11.00
image

1340
image

**Comment by SkullzOTS** [comment]
@kokekanon try changing it to fin->cache(true); on this line: https://github.com/mehah/otclient/blob/main/src/client/spriteappearances.cpp#L66

**Comment by kokekanon** [comment]
# 13.40

image
work

with fin->cache(true);

**Comment by SkullzOTS** [comment]
@kokekanon in version 13.40? Can you please also test if it works correctly when decrypted with this change?

**Comment by kokekanon** [comment]
the changes in this pr work

```cpp
      //  .\src\framework\config.h
        #define ENABLE_ENCRYPTION 1 + (encrypted files)
```
test in 
| Protocol 	| check encrypt|
|----------	|--------	|
| 860     	| ✅  	|  
| 1100     	| ✅		| 
| 1340     	| ✅  	|  


and does not cause problems in default (`#define ENABLE_ENCRYPTION 0 , non-encrypted files `)

**Comment by SkullzOTS** [comment]
> the changes in this pr work
> 
> ```c++
>       //  .\src\framework\config.h
>         #define ENABLE_ENCRYPTION 1 + (encrypted files)
> ```
> 
> test in
> 
> Protocol	check encrypt
> 860	✅
> 1100	✅
> 1340	✅
> and does not cause problems in default (`#define ENABLE_ENCRYPTION 0 , non-encrypted files `)

perfect... ty for your help! @kokekanon

File: src/framework/core/filestream.cpp [hunk 1]
```diff
@@ -85,7 +85,7 @@ void FileStream::cache(bool useEnc)
 
             uint8_t* decryptedData = ResourceManager::decrypt(m_data.data(), m_data.size());
             m_data.resize(size - std::string(ENCRYPTION_HEADER).size());
-            memcpy(m_data.data(), decryptedData, m_data.size());
+            std::memcmp(m_data.data(), decryptedData, m_data.size());
```
Comment (Kamenuvol) [inline src/framework/core/filestream.cpp:88]: Why compare memory and not assigning anywhere?

**Pull request merged**